### PR TITLE
:sparkles: New layer design

### PR DESCRIPTION
**Pull request by EvaMarco**
This PR adds a new components preview page and adds the new layers design

### CODE REVIEW

File: frontend/resources/images/icons/component-refactor.svg
Comment (Alotor) [inline frontend/resources/images/icons/component-refactor.svg:1]: This image is not well formed

File: frontend/resources/styles/common/refactor/basic-rules.scss [hunk 1]
```diff
@@ -0,0 +1,64 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) KALEIDOS INC
+
+.button-primary {
```
Comment (Alotor) [inline frontend/resources/styles/common/refactor/basic-rules.scss:7]: This file is all about buttons but it's called `basic-rules` maybe change the name to something more related to buttons?

File: frontend/resources/styles/common/refactor/common-refactor.scss [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+@import "common/refactor/borders.scss";
+@import "common/refactor/shadows.scss";
+@import "common/refactor/z-index.scss";
+@import "common/refactor/mixins.scss";
+@import "common/refactor/basic-rules.scss";
+
+::-webkit-scrollbar {
+  background-color: transparent;
+  cursor: pointer;
+  height: 8px;
```
Comment (Alotor) [inline frontend/resources/styles/common/refactor/common-refactor.scss]: shouldn't be `spacing-8`?

File: frontend/src/app/main/ui/components/shape_icon_refactor.cljs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+               i/flex-vertical-refactor
+
+               (and (ctl/flex-layout? shape) (ctl/row? shape))
+               i/flex-horizontal-refactor
+
+               ;; TODO: GRID ICON
+
+               :else
+               i/board-refactor) 
+      ;; TODO -> revisar el thumbnail y grid
```
Comment (Alotor) [inline frontend/src/app/main/ui/components/shape_icon_refactor.cljs]: change the comment to english

File: frontend/src/app/main/ui/debug/components_preview.scss [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+.themes-row {
+  width: 100%;
+  padding: 20px;
+  color: var(--color-foreground-primary);
+  background: var(--color-background-secondary);
+  .wrapper {
+    background-color: var(--color-background-primary);
+    width: 100%;
+    padding: 20px;
```
Comment (Alotor) [inline frontend/src/app/main/ui/debug/components_preview.scss]: in this file there are several `pixel` units, should we use the new `$s-X` to be more consistent?

File: frontend/src/app/main/ui/workspace/context_menu/context_menu.cljs [hunk 5]
```diff
@@ -75,43 +77,83 @@
          (fn [dom]
            (let [submenu-node (mf/ref-val submenu-ref)]
              (when (and (some? dom) (some? submenu-node))
-               (dom/set-css-property! submenu-node "top" (str (.-offsetTop dom) "px"))))))]
+               (dom/set-css-property! submenu-node "top" (str (.-offsetTop dom) "px"))))))
 
+        split-sc        (fn [sc]
```
Comment (Alotor) [inline frontend/src/app/main/ui/workspace/context_menu/context_menu.cljs]: I think this makes more sense in a funcion outside. It doesn't deppend of anything inside the component.

File: frontend/src/app/main/ui/workspace/context_menu/context_menu.scss [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+//
+// Copyright (c) KALEIDOS INC
+
+@import "refactor/common-refactor.scss";
+
+.workspace-context-menu,
+.workspace-context-submenu {
+  position: absolute;
+  top: $s-40;
+  left: 740px;
```
Comment (Alotor) [inline frontend/src/app/main/ui/workspace/context_menu/context_menu.scss]: This fixed value is a bit strange. Maybe can be calculated from others? also it's in pixels

File: frontend/src/app/main/ui/workspace/context_menu/context_menu.scss [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+        height: $s-20;
+        padding: $s-2 $s-6;
+        border-radius: $br6;
+        background-color: var(--menu-shortcut-background-color);
+      }
+    }
+
+    .submenu-icon {
+      position: absolute;
+      right: 1rem;
```
Comment (Alotor) [inline frontend/src/app/main/ui/workspace/context_menu/context_menu.scss]: $s-8?

File: frontend/src/app/main/ui/workspace/context_menu/context_menu.scss [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+    align-items: center;
+    height: $s-28;
+    padding: $s-6;
+    border-radius: $br8;
+    &:hover {
+      background-color: var(--menu-background-color-hover);
+    }
+
+    span.title {
+      margin-left: 5px;
```
Comment (Alotor) [inline frontend/src/app/main/ui/workspace/context_menu/context_menu.scss]: use the variable

File: frontend/src/app/main/ui/workspace/context_menu/context_menu.scss [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+
+    .selected-icon {
+      svg {
+        @extend .button-icon-small;
+        stroke: var(--menu-foreground-color);
+      }
+    }
+
+    .shape-icon {
+      margin-left: 3px;
```
Comment (Alotor) [inline frontend/src/app/main/ui/workspace/context_menu/context_menu.scss]: + variable

File: frontend/src/app/main/ui/workspace/sidebar/layers/layer_item/layer_item.scss [hunk 10]
```diff
@@ -0,0 +1,354 @@
...
+      .block-element {
+        svg {
+          stroke: var(--layer-row-component-foreground-color);
+        }
+      }
+    }
+    .element-children {
+      color: var(--layer-row-component-foreground-color);
+    }
+    &:gloabl(.hidden) {
```
Comment (Alotor) [inline frontend/src/app/main/ui/workspace/sidebar/layers/layer_item/layer_item.scss]: typo!

File: frontend/src/app/main/ui/workspace/sidebar/layers/layer_item/layer_item.scss [hunk 11]
```diff
@@ -0,0 +1,354 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (Alotor) [inline frontend/src/app/main/ui/workspace/sidebar/layers/layer_item/layer_item.scss:1]: This file has a lot of global styles... maybe shouldn't be inside a component?

File: frontend/resources/styles/common/refactor/basic-rules.scss (hunk 1 above)
Comment (EvaMarco) [reply frontend/resources/styles/common/refactor/basic-rules.scss:7]: As we talk this fill will contain more basic rules, for now I will keep that name. In the next PR I will check it and see if we need to change it's name.

**Pull request merged**